pkgs/tools/system/netdata/mosquitto.nix

```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchFromGitHub, c-ares, cmake, libwebsockets_3_2, openssl }:
+
+stdenv.mkDerivation rec {
+  pname = "mosquitto";
+  version = "1.6.8_Netdata-4";
+
+  src = fetchFromGitHub {
+    owner = "netdata";
+    repo = "mosquitto";
+    rev = "v.${version}";
+    sha256 = "06vb9pndw3zwfnh4a8rwafdkrz00kq01vngl0zq32y74h16pp2zb";
+  };
+
+  buildInputs = [ c-ares libwebsockets_3_2 openssl ];
+  nativeBuildInputs = [ cmake ];
+
+  cmakeFlags = [
+    "-DWITH_THREADING=YES"
+    "-DWITH_WEBSOCKETS=YES"
+    "-DWITH_STATIC_LIBRARIES=YES"
+    "-DWITH_DOCS=NO"
+  ];
+
+  buildPhase = ''
+    cd lib
+    make install
+  '';
+
+  postInstall = ''
+    cp $out/lib/libmosquitto_static.a $out/lib/libmosquitto.a
```

Comment:
Can we use ln here?

Reply:
Sorry, but I am not quite sure what you mean by "using ln", could you maybe elaborate?

Reply:
I think they mean doing something like this:
```
ln -s $out/lib/libmosquitto_static.a $out/lib/libmosquitto.a
```

Reply:
Apologies, `I` and `l` look the same to me in my system variable-width font, so I got confused. It does not seem possible to use symbolic links here (build fails) but hard links seem to be fine.

Reply:
I'm not sure if hard linking is something that is commonly done in Nixpkgs and whether it has any particular advantages or disadvantages. A very naive grep-based investigation shows that copying files is done much more frequently than hardlinking, so I'd probably just keep the `cp`.

```shell
$ cd nixpkgs
$ rg "cp[ ][^(-r)]" | wc -l
1451
$ rg "ln[ ][^(-s)]" | wc -l
29
```

Reply:
cp is useful when moving files from source to out but ln should be used when symlinking static libs in out or even mv.

Reply:
Ok, I have switched the two lines to hardlinking now.

> ln should be used when symlinking static libs in out or even mv

You are aware that we are not using symlinking here, right? I am now using hardlinking, which is a different concept which is why I am unsure if I should use it in derivations.